atomic_reactor/cli/parser.py

```diff
@@ -0,0 +1,132 @@
+    _add_git_source_args(orchestrator)
+
+    worker = tasks.add_parser(
+        "worker",
+        help="run the worker task",
+        description="Run the worker task for a binary container build.",
+    )
+    worker.set_defaults(func=task.worker)
+    _add_git_source_args(worker)
+
+    source_build = tasks.add_parser(
+        "source-build",
+        help="build a source container",
+        description="Build a source container.",
+    )
+    source_build.set_defaults(func=task.source_build)
+
+    return vars(parser.parse_args(args))
+
+
+def _add_global_args(parser: argparse.ArgumentParser) -> None:
+    """Add global arguments to the main parser."""
+    try:
+        version = pkg_resources.get_distribution("atomic_reactor").version
+    except pkg_resources.DistributionNotFound:
+        version = "GIT"
+
+    # -V/--version prints version info and exits the program
+    parser.add_argument("-V", "--version", action="version", version=version)
+    # -q/--quiet and -v/--verbose are exclusive for obvious reasons
+    verbosity_ex = parser.add_mutually_exclusive_group()
+    verbosity_ex.add_argument(
+        "-q",
+        "--quiet",
+        action="store_true",
+        help="suppress normal output, show only warnings and errors",
+    )
+    verbosity_ex.add_argument(
+        "-v",
+        "--verbose",
+        action="store_true",
+        help="be more verbose, include debug messages in output",
+    )
+
+
+def _add_common_task_args(task_parser: argparse.ArgumentParser) -> None:
+    """Add arguments common to all tasks to the task subparser."""
+    task_parser.add_argument(
+        "--build-dir",
+        metavar="DIR",
+        required=True,
+        help="directory for the build input files",
+    )
+    task_parser.add_argument(
+        "--context-dir",
+        metavar="DIR",
+        required=True,
+        help="shared working directory for tasks",
+    )
+    task_parser.add_argument(
+        "--config-file",
+        metavar="FILE",
+        default=REACTOR_CONFIG_FULL_PATH,
+        help=f"{PROG} configuration file",
+    )
+    # Two different ways to pass user params
+    userparams_ex = task_parser.add_mutually_exclusive_group()
+    userparams_ex.add_argument(
+        "--user-params", metavar="JSON", help="JSON string with user configuration"
+    )
+    userparams_ex.add_argument(
+        "--user-params-file", metavar="FILE", help="JSON file with user configuration"
+    )
+
+
+def _add_git_source_args(subtask_parser: argparse.ArgumentParser) -> None:
+    """Add --source-uri and --source-ref args to a specific task."""
+    subtask_parser.add_argument(
+        "--source-uri",
+        metavar="URI",
+        required=True,
+        help="git uri of the repository containing the build inputs",
+    )
+    subtask_parser.add_argument(
+        "--source-ref",
+        metavar="REF",
+        help="git ref of the repository containing the build inputs",
+    )
```

Comment:
also please add required args for source container

which is either sources-for-koji-build-nvr OR sources-for-koji-build-id  (ideally mutually exclusive, even though you can pass both, there is check in plugin to compare them anyway)